release/cockpit-release.yaml

```diff
@@ -0,0 +1,48 @@
+---
+apiVersion: v1
+kind: List
+items:
+- kind: Pod
+  apiVersion: v1
+  metadata:
+    name: release
+    labels:
+      infra: cockpit-release
+  spec:
+    containers:
+      - name: release
+        image: cockpit/release
+        ports:
+          - containerPort: 8080
+            protocol: TCP
+        command: [ "webhook" ]
+        volumeMounts:
+        - name: secrets
+          mountPath: /run/secrets/release
```

Comment:
Does this need container need access to all the secrets? Do you feel that the webhook script is sufficiently secure to prevent leaking the secrets? Should we have the webhook in its own container, which triggers a Kube/Openshift "Job" in order to do its work?

I know this is significantly different, and this problem may already exist with the current deployment.

Reply:
Thanks for pointing out [Jobs](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/)! In fact this is exactly what I was looking for, when googling something like "kubernetes service create pod on demand" (all existing hits I found just said "there is no such concept"). I don't particularly mistrust the HTTP server implementation, as it does not have any responses that are composed of file system items, but nevertheless this sounds like a better approach all around. I'm not very happy to have the release container run all the time - it should spin up, do *one* release, and get killed. `Job` is what I was missing :-)